api/v1alpha1/clusterextension_types.go

```diff
@@ -167,7 +159,8 @@ type BundleMetadata struct {
 	Version string `json:"version"`
 }
 
-// ClusterExtensionStatus defines the observed state of ClusterExtension
+// ClusterExtensionStatus defines the observed state of ClusterExtension.
+// InstalledBundle should only be modified when a new bundle is successfully installed. This ensures that if you've previously successfully installed a bundle before, and an upgrade fails, it is still appropriately communicated to you that there is still a bundle that is currently installed and owned by the ClusterExtension.
```

Comment:
Two comments:
1. I don't think the `Status` object is the right place for details of the `InstalledBundle` field. Move this comment to that field?
2. We should be able to assess a "current installation" independent of a bunch of other reconcile tasks. Our reconcile behavior probably needs to go and actually lookup the currently installed bundle early on and affirmatively set `status.conditions[type=Installed]` and `status.installedBundle` based on what we find. If we fail trying to lookup the installed bundle, we should set the condition to Unknown, probably unset the `installedBundle` field, and then continue on with other unrelated reconcile tasks.

Reply:
> I don't think the Status object is the right place for details of the InstalledBundle field. Move this comment to that field?

@joelanford Not sure if I understand your comments fully. Currently `InstalledBundle` is part of ClusterExtensionStatus , I guess you are suggesting it should not?

```
type ClusterExtensionStatus struct {
	// +optional
	InstalledBundle *BundleMetadata `json:"installedBundle,omitempty"`
	// +optional
	ResolvedBundle *BundleMetadata `json:"resolvedBundle,omitempty"`
	// +patchMergeKey=type
	// +patchStrategy=merge
	// +listType=map
	// +listMapKey=type
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
}
```

Reply:
Also I do not see any suggested change for the `InstalledBundle` in the RFC document.

Reply:
Fixing (1) above
on point (2), I'd llke to add https://github.com/operator-framework/operator-controller/issues/1012 and keep this PR focused on implementing the RFC.

Reply:
@LalatenduMohanty InstalledBundle handing is getting changed here: https://github.com/operator-framework/operator-controller/blob/main/internal/controllers/common_controller.go#L111 to partially answer @joelanford and others comments about when and how we should set it, so yes, not in the RFC, hence the issue add and I'd like to deal with it more thoroughly outside this PR. I wanted to add a better comment to the field here to reflect how we intend its usage. Joe's comment is simply I think that he'd like to see the comment down by the field itself, not up on the struct, which I'll take care of shortly.